### PR TITLE
Add timestamps to comments proposal

### PR DESCRIPTION
**Pull request by alexr00**
First crack at adding timestamps to comments.
Part of #139524

### CODE REVIEW

**Comment by alexr00** [comment]
For now, I've gone with an API where the extension can opt in to using a relative date. The displayed relative date will not be updated so long as the editor is visible, but when the editor is not visible and then made visible again, it does get re-displayed, and thus updated. We can revisit this and as long as the editor is visible do some periodic update.

For date display I've used dayjs. I have not tested with other locales, but I am using the locale friendly way of showing the long date on hover/when no relative date is used, so I would expect it to work properly for other locales.

**Comment by alexr00** [comment]
Updates to the Comments view are not included in this PR and will come later.

File: .eslintrc.json [hunk 1]
```diff
@@ -568,7 +568,9 @@
 					"vscode-oniguruma",
 					"iconv-lite-umd",
 					"tas-client-umd",
-					"jschardet"
+					"jschardet",
```
Comment (rebornix) [inline .eslintrc.json:571]: Maybe you have already discussed this in the API Sync, but FYI that the timeline api uses timestamp https://github.com/microsoft/vscode/blob/a4d426a1c2ece4222f1343459d31fe948098b2f6/src/vscode-dts/vscode.proposed.timeline.d.ts#L14, not sure if we wanna keep it aligned.

**Review by rebornix — APPROVED** [review]
The changes to the commenting code look good to me. My only comments are related to the shape of the API, please ignore if they were already discussed in API meetings:

* Timeline API uses timestamp, maybe we want to stay aligned
* `detail` seems ambiguous as it can be used for both date time and raw text

**Comment by alexr00** [comment]
> Timeline API uses timestamp, maybe we want to stay aligned

Using `number` instead of `Date` to align with timeline makes sense.

> detail seems ambiguous as it can be used for both date time and raw text

I actually started with `timestamp` but then during API discussion `detail` was preferred because it allows flexibility and is consistent with other places we use `detail`.

**Review by raghavthind2005 — APPROVED** [review]

**Comment by alexr00** [comment]
Outcome from API discussion:
- We will stick with `detail` as this is a familiar concept elsewhere.
- We will stick with `Date` instead of `number` since it is more descriptive. Additionally, we should consider aligning timeline to use `Date` as well.

**Pull request merged**